assets/js/components/surveys/CurrentSurvey.test.js

```diff
@@ -0,0 +1,412 @@
+
+		fetchMock.post( /^\/google-site-kit\/v1\/core\/user\/data\/survey-event/, { body: {}, status: 200 } );
+
+		const { getByLabelText } = render( <CurrentSurvey />, { registry } );
+
+		fireEvent.click( getByLabelText( 'Delighted icon' ) );
+
+		expect(
+			registry
+				.select( CORE_FORMS )
+				.getValue(
+					`survey-${ fixtures.singleQuestionSurvey.session.session_id }`,
+					'answers'
+				)
+		).toEqual(
+			[ { question_ordinal: 1, answer_ordinal: 5 } ]
+		);
+	} );
+
+	it( 'should render nothing if no survey exists', () => {
+		const { container } = render( <CurrentSurvey />, { registry } );
+
+		expect( container ).toBeEmptyDOMElement();
+	} );
+
+	it( "should send a 'survey_closed' event when dismissed", () => {
+		registry.dispatch( CORE_USER ).receiveTriggerSurvey( fixtures.singleQuestionSurvey, { triggerID: 'jestSurvey' } );
+
+		fetchMock.post( /^\/google-site-kit\/v1\/core\/user\/data\/survey-event/, { body: {}, status: 200 } );
+
+		const { getByLabelText } = render( <CurrentSurvey />, { registry } );
+
+		fireEvent.click( getByLabelText( 'Dismiss this survey' ) );
+
+		expect( fetchMock ).toHaveFetched(
+			'/google-site-kit/v1/core/user/data/survey-event?_locale=user',
+			{
+				body: {
+					data: {
+						event: {
+							survey_closed: {},
+						},
+						session: fixtures.singleQuestionSurvey.session,
+					},
+				},
+				credentials: 'include',
+				headers: {
+					Accept: 'application/json, */*;q=0.1',
+					'Content-Type': 'application/json',
+				},
+				method: 'POST',
+			}
+		);
+	} );
+
+	it( 'should render nothing if the survey is dismissed', () => {
+		registry.dispatch( CORE_USER ).receiveTriggerSurvey( fixtures.singleQuestionSurvey, { triggerID: 'jestSurvey' } );
+
+		fetchMock.post( /^\/google-site-kit\/v1\/core\/user\/data\/survey-event/, { body: {}, status: 200 } );
+
+		const { container, getByLabelText } = render( <CurrentSurvey />, { registry } );
+
+		fireEvent.click( getByLabelText( 'Dismiss this survey' ) );
+
+		expect( container ).toBeEmptyDOMElement();
+	} );
+
+	it( 'should render the completed survey component if all questions have been answered', () => {
+		registry.dispatch( CORE_USER ).receiveTriggerSurvey( fixtures.singleQuestionSurvey, { triggerID: 'jestSurvey' } );
+
+		registry.dispatch( CORE_FORMS ).setValues(
+			`survey-${ fixtures.singleQuestionSurvey.session.session_id }`,
+			{
+				// Mark this survey as answered so the "Survey Complete" component is
+				// rendered.
+				answers: [
+					{ question_ordinal: 1, answer_ordinal: 2 },
+				],
+			}
+		);
+
+		fetchMock.post( /^\/google-site-kit\/v1\/core\/user\/data\/survey-event/, { body: {}, status: 200 } );
+
+		const { container } = render( <CurrentSurvey />, { registry } );
+
+		expect( container ).toMatchSnapshot();
+	} );
+
+	it( 'should trigger a completion once all questions are answered, even if no matching trigger_conditions are found', () => {
```

Comment:
This could maybe be named more explicitly?
```suggestion
	it( 'should trigger the first completion when no matching trigger_conditions are met', () => {
```